src/validate.rs

```diff
@@ -10,14 +10,361 @@ use crate::base::name::Name;
+        data: &[u8],
+    ) -> Result<Self, FromDnskeyError> {
+        match algorithm {
+            SecAlg::RSASHA1 => {
+                RsaPublicKey::from_dnskey(data).map(Self::RsaSha1)
+            }
+            SecAlg::RSASHA1_NSEC3_SHA1 => {
+                RsaPublicKey::from_dnskey(data).map(Self::RsaSha1Nsec3Sha1)
+            }
+            SecAlg::RSASHA256 => {
+                RsaPublicKey::from_dnskey(data).map(Self::RsaSha256)
+            }
+            SecAlg::RSASHA512 => {
+                RsaPublicKey::from_dnskey(data).map(Self::RsaSha512)
+            }
+
+            SecAlg::ECDSAP256SHA256 => {
+                let mut key = Box::new([0u8; 65]);
+                if key.len() == 1 + data.len() {
+                    key[0] = 0x04;
+                    key[1..].copy_from_slice(data);
+                    Ok(Self::EcdsaP256Sha256(key))
+                } else {
+                    Err(FromDnskeyError::InvalidKey)
+                }
+            }
+            SecAlg::ECDSAP384SHA384 => {
+                let mut key = Box::new([0u8; 97]);
+                if key.len() == 1 + data.len() {
+                    key[0] = 0x04;
+                    key[1..].copy_from_slice(data);
+                    Ok(Self::EcdsaP384Sha384(key))
+                } else {
+                    Err(FromDnskeyError::InvalidKey)
+                }
+            }
+
+            SecAlg::ED25519 => Box::<[u8]>::from(data)
+                .try_into()
+                .map(Self::Ed25519)
+                .map_err(|_| FromDnskeyError::InvalidKey),
+            SecAlg::ED448 => Box::<[u8]>::from(data)
+                .try_into()
+                .map(Self::Ed448)
+                .map_err(|_| FromDnskeyError::InvalidKey),
+
+            _ => Err(FromDnskeyError::UnsupportedAlgorithm),
+        }
+    }
+
+    /// Parse a public key from a DNSKEY record in presentation format.
+    ///
+    /// This format is popularized for storing alongside private keys by the
+    /// BIND name server.  This function is convenient for loading such keys.
+    ///
+    /// The text should consist of a single line of the following format (each
+    /// field is separated by a non-zero number of ASCII spaces):
+    ///
+    /// ```text
+    /// <domain-name> <record-class> DNSKEY <record-data> [<comment>]
+    /// ```
+    ///
+    /// Where `<record-data>` consists of the following fields:
+    ///
+    /// ```text
+    /// <flags> <protocol> <algorithm> <encoded-public-key>
+    /// ```
+    ///
+    /// The first three fields are simple integers, while the last field is
+    /// Base64 encoded data (with or without padding).  The [`from_dnskey()`]
+    /// and [`to_dnskey()`] read from and serialize to the Base64-decoded data
+    /// format.
+    ///
+    /// [`from_dnskey()`]: Self::from_dnskey()
+    /// [`to_dnskey()`]: Self::to_dnskey()
+    ///
+    /// The `<comment>` is any text starting with an ASCII semicolon.
+    pub fn from_dnskey_text(
```

Comment:
Point of consistency: we have `SecretKey::parse_from_bind()` and `PublicKey::from_dnskey_text()` - why is one called parse and the other is not?

Reply:
Right, will rename.